### DISCUSSION TITLE
scraping

### DISCUSSION THREAD

**Discussion by AKSHITHA-CHILUKA** [recodehive/Scrape-ML#133] [Q&A]
A part from selenium what scraping tools can I use

**Comment by RaghavSingh1212** [comment]
Apart from Selenium, you can use **BeautifulSoup** for simple HTML parsing on static pages, **Scrapy** for efficient large-scale scraping, and **Playwright** as a modern, reliable alternative for handling dynamic content. **Puppeteer** is great if you prefer JavaScript and need to scrape Chromium-based pages. For lightweight scraping, **requests + lxml** work well for static sites. If you need a scalable, cloud-based solution, **Apify** offers automation and headless browsing capabilities.

**Comment by Svintjohn** [comment]
If you don’t wanna use Selenium, you’ve got other chill options. For simple pages, just use Requests + BeautifulSoup, super easy. If the site is full of JavaScript, Playwright or Puppeteer works way better. For big scraping projects, Scrapy is solid. And if you're on Node.js, Cheerio is quick and clean. Just pick whatever fits the site you’re dealing with.
